drivers/regulator/regulator_cp9314.c

```diff
@@ -277,6 +279,23 @@ static int regulator_cp9314_enable(const struct device *dev)
 		return ret;
 	}
 
+#ifdef CONFIG_MULTITHREADING
+	k_usleep(CP9314_T_STARTUP_MSEC);
+#else
+	k_busy_wait(CP9314_T_STARTUP_MSEC);
+#endif
```

Comment:
unless I'm missing something, `k_usleep` already takes care of figuring out whether multi-threading is enable or not?

Reply:
I am just surprised that the #ifdef linked in the comment you're mentioning seems to be literally the only place in-tree where such k_sleep vs. k_busy_wait ifdef'ery is happening. But there might very well be a reason that I am completely missing and Andy will clarify :)

Reply:
Regulators are required to work when `CONFIG_MULTITHREADING=n` so this was necessary to avoid build failure.
However I was not aware that it was fixed here:
https://github.com/zephyrproject-rtos/zephyr/pull/63994

So it is no longer necessary :-)

Reply:
Thanks @aasinclair and @kartben! I'll make sure to clean up the rest of `regulator_cp9314.c` in subsequent PR soon.